tt_metal/tt_metal.cpp

```diff
@@ -1137,37 +1150,38 @@ std::unique_ptr<GlobalSemaphore> CreateGlobalSemaphore(
     return GlobalSemaphore::create(device, std::move(cores), initial_value, buffer_type);
 }
 
-std::shared_ptr<Buffer> CreateBuffer(const InterleavedBufferConfig &config) {
-    return Buffer::create(
-        config.device, config.size, config.page_size, config.buffer_type, config.buffer_layout, std::nullopt, std::nullopt);
-}
-
-std::shared_ptr<Buffer> CreateBuffer(const InterleavedBufferConfig &config, DeviceAddr address) {
-    return Buffer::create(
-        config.device, address, config.size, config.page_size, config.buffer_type, config.buffer_layout, std::nullopt, std::nullopt);
-}
-
-std::shared_ptr<Buffer> CreateBuffer(const ShardedBufferConfig &config) {
-    return Buffer::create(
-        config.device,
-        config.size,
-        config.page_size,
-        config.buffer_type,
-        config.buffer_layout,
-        config.shard_parameters,
-        std::nullopt);
+std::shared_ptr<Buffer> CreateBuffer(const InterleavedBufferConfig &config, std::optional<DeviceAddr> address, std::optional<uint32_t> sub_device_id) {
```

Comment:
Please don't combine these overloads, I'd rather avoid the unnecessary `std::optional` branching here, where the logical was previously unconditional.

Reply:
Do you have a suggestion for avoiding this? I ended up combining them because the uint64_t address and The uint32_t sub_device_id are both integer type, so it was ambiguous to overload.

Reply:
Make `sub_device_id` a strong type. `DeviceAddr` should have been a strong type too. Making them aliases for numeric types is what leads to completely avoidable issues like this.